tajo-catalog/tajo-catalog-client/src/main/java/org/apache/tajo/catalog/AbstractCatalogClient.java

```diff
@@ -574,25 +652,26 @@ public final FunctionDesc getFunction(final String signature, FunctionType funcT
       builder.addParameterTypes(type);
     }
 
-    FunctionDescProto descProto = null;
+    FunctionDescResponse response = null;
     try {
-      CatalogProtocolService.BlockingInterface stub = getStub();
-      descProto = stub.getFunctionMeta(null, builder.build());
-    } catch (NoSuchFunctionException e) {
-      LOG.debug(e.getMessage());
-    } catch (ServiceException e) {
-      LOG.error(e.getMessage(), e);
+      final BlockingInterface stub = getStub();
+      response = stub.getFunctionMeta(null, builder.build());
+    } catch (ServiceException se) {
+      throw new RuntimeException(se);
     }
 
-    if (descProto == null) {
-      throw new NoSuchFunctionException(signature, paramTypes);
+    if (isThisError(response.getState(), ResultCode.UNDEFINED_FUNCTION)) {
+      throw new UndefinedFunctionException(signature, paramTypes);
+    } else if (isThisError(response.getState(), ResultCode.AMBIGUOUS_FUNCTION)) {
```

Comment:
There is a mismatch. DuplicateFunctionException should be AmbiguousFunctionException. It's my intension. 'Duplicate function exception' will occur only at CREATE FUNCTION.